src/api/endpoints/balance/history.ts

```diff
@@ -27,51 +62,65 @@ export const getHistory: RouteOptions = {
     const query = request.query as HistoryQuery;
     const days = getDaysDifference(query);
     const address = toBuffer(query.address);
-    const history: any | null = await redb.manyOrNone(
-      `select uav2.contract_address, uav2.wallet_address,  uav2.total_recieve/power(10, awp.decimals) as total_recieve,
-      total_transfer/power(10, awp.decimals) as total_transfer, total_amount/power(10, awp.decimals) as total_amount, awp."name" "contractName",
-      awp.coingecko_id as "coingeckoId", transfer_count, receive_count, awp.price as usd_price  
-      from user_activity_view uav2 right join assets_with_price awp on uav2.contract_address = awp.contract and uav2."timestamp" <= awp."timestamp" where uav2."timestamp" 
+    const history: RawHistoryObject[] | null = await redb.manyOrNone(
+      `select SUM(y.total_amount) total_amount, SUM(y.usd_price) usd_price, y.wallet_address, y.timestamp from (
+        select uav2.timestamp, uav2.wallet_address, total_amount/power(10, awp.decimals) as total_amount, awp.price * total_amount/power(10, awp.decimals) as usd_price
+        from user_activity_view uav2 right join assets_with_price awp on uav2.contract_address = awp.contract and uav2."timestamp" <= awp."timestamp" where uav2."timestamp" 
```

Comment:
`user_activity_view` is the table. Do you think we can make it a view?
If that's not possible, then at least let's rename the table.

Reply:
We created `user_activity_view` to show history, why we need this computation here?

Reply:
rename can be done,
commutation it just to get the actual value, after dividing them with decimal of that asset

Reply:
no, I'm talking about sub query and run time joins